pkg/operator2/routercerts/controller_test.go

```diff
@@ -0,0 +1,365 @@
+				withPEM(
+					withServer("*.example.com",
+						withCA(
+							withCA(),
+						),
+					),
+				),
+			)),
+		},
+		{
+			name:           "RootCAIsSystemCert",
+			ingress:        newIngress("cluster", withDomain("example.com")),
+			expectedStatus: operatorv1.ConditionFalse,
+			expectedReason: "AsExpected",
+			secret: newSecret("router-certs", "target", withData("example.com",
+				withPEM(
+					withServer("*.example.com",
+						withCA(
+							withCA(
+								withExistingRootCA(fakeSystemRootCA),
+							),
+						),
+						withMissingRootCA,
+					),
+				),
+			)),
+			systemCertPool: func() (*x509.CertPool, error) { return fakeSystemCertPool, nil },
+		},
+		{
+			name:           "NoIngressConfig",
+			expectedStatus: operatorv1.ConditionTrue,
+			expectedReason: "NoIngressConfig",
+			secret:         newSecret("router-certs", "target"),
+		},
+		{
+			name:           "NoIngressDomain",
+			expectedStatus: operatorv1.ConditionTrue,
+			expectedReason: "NoIngressDomain",
+			ingress:        newIngress("cluster"),
+			secret:         newSecret("router-certs", "target"),
+		},
+		{
+			name:           "NoRouterCertSecret",
+			expectedStatus: operatorv1.ConditionTrue,
+			expectedReason: "NoRouterCertSecret",
+			ingress:        newIngress("cluster", withDomain("example.com")),
+		},
+		{
+			name:           "MissingRouterCertsPEM",
+			expectedStatus: operatorv1.ConditionTrue,
+			expectedReason: "MissingRouterCertsPEM",
+			ingress:        newIngress("cluster", withDomain("example.com")),
+			secret:         newSecret("router-certs", "target"),
+		},
+		{
+			name:           "MalformedRouterCertsPEM",
+			ingress:        newIngress("cluster", withDomain("example.com")),
+			secret:         newSecret("router-certs", "target", withData("example.com")),
+			expectedStatus: operatorv1.ConditionTrue,
+			expectedReason: "MissingRouterCertsPEM",
+		},
+		{
+			name:           "NoRootCARouterCerts",
+			expectedStatus: operatorv1.ConditionTrue,
+			expectedReason: "NoRootCARouterCerts",
+			ingress:        newIngress("cluster", withDomain("example.com")),
+			secret: newSecret("router-certs", "target", withData("example.com",
+				withPEM(
+					withServer("*.example.com",
+						withCA(
+							withCA(),
+						),
+						withMissingRootCA,
+					),
+				),
+			)),
+			systemCertPool: func() (*x509.CertPool, error) { return x509.NewCertPool(), nil },
+		},
+		{
+			name:    "NoServerCertRouterCerts",
+			ingress: newIngress("cluster", withDomain("example.com")),
+			secret: newSecret("router-certs", "target", withData("example.com",
+				withPEM(
+					withCA(
+						withCA(),
+					),
+				),
+			)),
+			expectedStatus: operatorv1.ConditionTrue,
+			expectedReason: "NoServerCertRouterCerts",
+		},
+		{
+			name:           "InvalidServerCertRouterCerts",
+			ingress:        newIngress("cluster", withDomain("example.com")),
+			expectedStatus: operatorv1.ConditionTrue,
+			expectedReason: "InvalidServerCertRouterCerts",
```

Comment:
I would expect to see a few variations on this particular failure - like wrong DNS name, has roots but does not verify, etc.